### DISCUSSION TITLE
Multi training set/enhance (x,y) symmetry

### DISCUSSION THREAD

**Discussion by SimonYYS** [dynamicslab/pysindy#439] [General]
We would like to derive PDE equations from a particle simulation. Our 2D system (PDEs, an active nematic system like liquid crystal) is sensitive to _(x,y)_ coordinate, and in principal, the output equations should be symmetric in **x** and **y**. For instance, the velocity is an vector in the form of (vx, vy), and I separate it into 2 equations in a bid to reduce to scaler form (not sure if this is the best way actually). Thus, the two equations of **vx** and **vy** should have the same (number of) terms with symmetry in **x** and **y**, which is not enforced now.

However, when generating training set from a particle simulation, there may be bias in **x** or **y** direction due to random initial condition and the output is not symmetric for **vx** and **vy**. I would appreciate it so much if anyone could give suggestions! Is it possible to fed the algorithm with multi training set to avoid this bias?

**Comment by Jacob-Stevens-Haas** [comment]
Cool use case!

> the output equations should be symmetric in x and y.

Sounds like a good case for `ConstrainedSR3` optimizer!  You'll have to write constraints for the symmetric terms

> Is it possible to fed the algorithm with multi training set to avoid this bias?

Yes, just pass a list of arrays for `x` and they will all be used for training.  For PDEs they all need to be on the same spatial domain, currently.

**Reply by SimonYYS** [reply]
Thanks for the reply! I will check **ConstrainedSR3** tutorial carefully.

> Yes, just pass a list of arrays for x and they will all be used for training. For PDEs they all need to be on the same spatial domain, currently.

A direct linking of two training sets may cause trouble in calculating time derivative ( **t = t_max** followed by a novel **t = 1** ). So I guess I need to link them after calculating of time derivative separately. Basically, I am not sure whether the time derivative between the ending of one training set and the begining of another could directly be connected into a larger array or not...

**Reply by Jacob-Stevens-Haas** [reply]
We calculate the derivatives _before_ concatenating the arrays.  So the time derivative in one sample is not affected by the data in another sample.

Also note that unless you're using the pysindy `master` branch, you need to pass `multiple_trajectories=True` to `SINDy.fit()` when passing multiple trajectories.
